amplify/backend/function/fetchDomainBalance/src/api/graphql/schemas.js

```diff
@@ -0,0 +1,174 @@
+const tokenFragment = /* GraphQL */ `
+  fragment LightTokenFragment on Token {
+    chainMetadata {
+      chainId
+      network
+    }
+    createdAt
+    decimals
+    id
+    name
+    symbol
+    type
+    updatedAt
+  }
+`;
+
+module.exports = {
+  saveTokenExchangeRate: /* GraphQL */`
+    mutation SaveTokenExchangeRate($input: CreateTokenExchangeRateInput!) {
+      createTokenExchangeRate(input: $input) {
+        tokenId
+        date
+        marketPrice {
+            currency
+            rate
+          }
+      }
+    }
+  `,
+  getTokenExchangeRate: /* GraphQL */`
+    query GetTokenExchangeRateByTokenId(
+      $tokenId: String!
+      $date: String
+      $filter: ModelTokenExchangeRateFilterInput
+    ) {
+      tokenExhangeRateByTokenId(
+        tokenId: $tokenId
+        date: { eq: $date }
+        filter: $filter
+      ) {
+        items {
+          tokenId
+          date
+          marketPrice {
+            currency
+            rate
+          }
+        }
+        nextToken
+      }
+    }
+  `,
+  getColonyDomains: /* GraphQL */`
+    query GetColonyDomains(
+      $colonyAddress: ID!
+    ) {
+      listDomains(
+        filter: {
+          colonyId: { eq: $colonyAddress }
+        }
+      ) {
+        items {
+          nativeFundingPotId
+          nativeId
+          colonyId
+          id
+          isRoot
+          # probably we need to rootId when nested teams is in place
+        }
+      }
+    }
+  `,
+  getColonyFundsClaims: /* GraphQL */`
+    ${tokenFragment}
+    query GetColonyFundsClaims(
+      $colonyAddress: ID!
+      $nextToken: String
+      $limit: Int
+    ) {
+      listColonyFundsClaims(
+        nextToken: $nextToken
+        limit: $limit
+        filter: {
+          isClaimed: { eq: true },
+          colonyFundsClaimsId: { eq: $colonyAddress }
+        }
+      ) {
+        items {
+          amount
+          isClaimed
+          updatedAt
+          colonyFundsClaimsId
+          token {
+            ...LightTokenFragment
+          }
+        }
+      }
+    }
+  `,
+  getDomainExpenditures: /* GraphQL */`
+    query GetDomainExpenditures(
+      $nativeDomainId: Int!
+      $nextToken: String
+      $limit: Int
+    ) {
+      listExpenditures(
+        nextToken: $nextToken
+        limit: $limit
+        filter: {
+          nativeDomainId: { eq: $nativeDomainId },
+          status: { eq: FINALIZED }
+        }
+      ) {
+        items {
+          id
+          finalizedAt
+          slots {
+            claimDelay
+            id
+            payouts {
+              amount
+              isClaimed
+              networkFee
+              tokenAddress
+            }
+          }
+        }
+      }
+    }
+  `,
```

Comment:
Getting expenditures by domain is a bit tricky - `nativeDomainId` denotes the domain in which the expenditure was created (relevant for permission checks) and not the domain it was funded from. In fact, an expenditure could be funded from many domains. 

Instead, you could query for MOVE_FUNDS actions that have an expenditureId set - that way you can tell which domain the expenditure has been funded from.

Reply:
@jakubcolony I created a new issue based on your insightful explanations from last week
https://github.com/JoinColony/colonyCDapp/issues/3067